src/insidersUpdate/updateCheck.ts

```diff
@@ -0,0 +1,301 @@
+        }
+    }
+
+    return undefined;
+}
+
+/**
+ * Downloads the VSIX.
+ * @param asset The GitHub asset to download.
+ */
+async function tryDownloadVsix(asset: GitHubAsset): Promise<vscode.Uri | undefined> {
+    const vsixDownloadUri: vscode.Uri = vscode.Uri.file(Utilities.generateTempPath(vsixAssetName));
+
+    // Save VSCode's proxy support setting. This is done because
+    // we attempt two download the VSIX with and without VSCode's
+    // proxy support.
+    const config: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration();
+    const originalProxySupport: string | undefined = config.inspect<string>('http.proxySupport')?.globalValue;
+    let restoreVSCodeProxySetting: boolean = false;
+
+    try {
+        let tryAgain: boolean = true;
+        while (tryAgain) {
+            try {
+                await downloadOverHttps(vscode.Uri.parse(asset.browser_download_url, /*strict*/ true), vsixDownloadUri);
+                return vsixDownloadUri;
+            } catch {
+                // If we failed to download, then turn off VSCode's proxy support (if it's not already off)
+                // and try again.
+                if (config.get('http.proxySupport', undefined) !== 'off' && originalProxySupport !== 'off') {
+                    await config.update('http.proxySupport', 'off', true);
+                    restoreVSCodeProxySetting = true;
+                    continue;
+                }
+
+                tryAgain = false;
+            }
+        }
+    } finally {
+        if (restoreVSCodeProxySetting) {
+            await config.update('http.proxySupport', originalProxySupport, true); // Reset the http.proxySupport.
+        }
+    }
+
+    return undefined;
+}
+
+/**
+ * Checks GitHub for an insiders update.
+ */
+export async function checkForInsiderUpdates(installOptions: 'Install' | 'Just check'): Promise<boolean> {
+    // If we have exceeded the rate limit (which actually would be hard to do since the limit is 60 per hour per IP
+    // for unauthorized requests), then we can't really do much.
+    // https://developer.github.com/v3/#rate-limiting
+    const gitHubRateLimit: GitHubRateLimitResponse  = await downloadOverHttpsAsJsonObject(vscode.Uri.parse(`${GitHubApiBase}/rate_limit`), 'rateLimit.json', userAgentHeaders);
+    if (gitHubRateLimit.resources.core.remaining <= 0) {
+        return false;
+    }
+
+    const gitHubRelease: GitHubRelease | undefined = await findInsidersReleaseCandidate();
+    if (!gitHubRelease) {
+        return false;
+    }
+
+    const gitHubAsset: GitHubAsset | undefined = await findAssetForRelease(gitHubRelease);
+    if (!gitHubAsset) {
+        return false;
+    }
+
+    const vsixUri: vscode.Uri | undefined = await tryDownloadVsix(gitHubAsset);
+    if (!vsixUri) {
+        return false;
+    }
+
+    // Since everything is highly async and it can take a while to download the VSIX, make sure one
+    // more time that the user is on the insiders version.
```

Comment:
This comment is misleading to me. I think the point is: "The download can take a while. The user can change the "update channel" setting during this time. Thus, we need to check one more time."

Reply:
If you concerned about that, then if they toggled the switch off/on during this time, it could cause multiple simultaneous downloads/updates correct?

Reply:
English is my 3rd language, especially grammar.

---
In reply to: [416055759](https://github.com/microsoft/sarif-vscode-extension/pull/267#discussion_r416055759) [](ancestors = 416055759)

Reply:
Yep. Correct. I should have handled that. Added code for that.

---
In reply to: [416057404](https://github.com/microsoft/sarif-vscode-extension/pull/267#discussion_r416057404) [](ancestors = 416057404)